lib/saml/ssoe_settings_service.rb

```diff
@@ -41,6 +41,26 @@ def base_settings
       def pki_needed?
         Settings.saml_ssoe.request_signing || Settings.saml_ssoe.response_encryption
       end
+
+      private
+
+      def saml_ssoe_certificate
+        if Settings.saml_ssoe.cert_path.present? && File.file?(File.expand_path(Settings.saml_ssoe.cert_path))
+          File.read(File.expand_path(Settings.saml_ssoe.cert_path))
+        end
+      end
+
+      def saml_ssoe_certificate_new
+        if Settings.saml_ssoe.cert_new_path.present? && File.file?(File.expand_path(Settings.saml_ssoe.cert_new_path))
+          File.read(File.expand_path(Settings.saml_ssoe.cert_new_path))
+        end
+      end
+
+      def saml_ssoe_key
+        if Settings.saml_ssoe.key_path.present? && File.file?(File.expand_path(Settings.saml_ssoe.key_path))
+          File.read(File.expand_path(Settings.saml_ssoe.key_path))
+        end
+      end
```

Comment:
I was thinking about what caching strategy we would use for this, and I'm not sure what the best way forward is. The certs aren't stored in a model, so we can't use `updated_at` to bust the cache. We could do a time-based expiration if we are OK with the old certificates being read for a period of time before the cache expires and the new certs are fetched.

I'm not 100% sure what the cert rotation looks like, but I think the time-based cache expiration would work. Is that what you had in mind?

Reply:
Yes the cache mechanism was discussed initially. But in later discussions we talked about not using the cache yet. I had the cache mechanism working but I removed it after the discussion of not using the cache. 

Yes I can dry those methods.

Reply:
Overall, the reason for moving this code was not to make things more efficient, but to prevent issues with Settings reloading and losing the changes made in the initializer, I'm not sure if other teams mutate the Settings file, but we probably shouldn't do it.

The reason for leaving out the cache in the initial version is that all services (including this one), already read from files whenever it's called, so it makes the first change simple just to move the logic over. It doesn't seem like it will make much of a change to compute or IO and will prevent us from breaking down if someone refreshes the Settings object